packages/button/src/Button.ts

```diff
@@ -37,17 +54,51 @@ export class Button extends SizedMixin(StyledButton) {
      * The visual variant to apply to this button.
      */
     @property({ reflect: true })
-    public variant: ButtonVariants = 'cta';
+    public get variant(): ButtonVariants {
+        return this._variant;
+    }
+    public set variant(variant: ButtonVariants) {
+        if (variant === this.variant) return;
+
+        this.requestUpdate('variant', this.variant);
+        switch (variant) {
+            case 'cta':
+                this._variant = 'accent';
+                break;
+            case 'overBackground':
+                this._variant = 'white';
+                this.treatment = 'outline';
+                break;
+            default:
+                if (!VALID_VARIANTS.includes(variant)) {
+                    this._variant = 'accent';
+                } else {
+                    this._variant = variant;
+                }
+                break;
+        }
+        this.setAttribute('variant', this.variant);
```

Comment:
`reflect: true` doesn't actually work when using custom getter/setters, so I've removed it. If there were secondary side effects `this.treatment = 'outline';` this would likely be better served with a custom `converter`. Hopefully we don't keep support  for these deprecated values for too long and this can go back to what it was, soon.